index.bs

```diff
@@ -261,16 +282,33 @@ dictionary MediaStreamTrackGeneratorInit {
     with |init|.{{MediaStreamTrackGeneratorInit/kind}}.
 4. Return |g|.
 
-### Attributes ### {#attributes-generator}
+### Attributes ### {#generator-attributes}
 <dl>
 <dt><dfn attribute for=MediaStreamTrackGenerator>writable</dfn></dt>
-<dd>Allows writing media frames the {{MediaStreamTrackGenerator}}, which is
-itself a {{MediaStreamTrack}}. If the {{MediaStreamTrack/kind}} attribute is `"audio"`,
-the stream will accept {{AudioData}} objects and fail with any other type. If
-{{MediaStreamTrack/kind}} is `"video"`, the stream will accept {{VideoFrame}} objects
-and fail with any other type. When a frame is written to {{MediaStreamTrackGenerator/writable}},
-the frame's `close()` method is automatically invoked, so that its internal resources are no longer
-accessible from JavaScript.
+<dd>Allows writing media frames to the {{MediaStreamTrackGenerator}}, which is
+itself a {{MediaStreamTrack}}. When this attribute is access for the first time,
+it MUST be initialized with the following steps:
+1. Initialize [=this=].{{MediaStreamTrackGenerator/writable}} to be a new {{WritableStream}}.
+2. <a dfn for="WritableStream">Set up</a> [=this=].{{MediaStreamTrackGenerator/writable}}, with its [=WritableStream/set up/writeAlgorithm=] set to [=writeFrame=] with [=this=] as parameter, [=WritableStream/set up/closeAlgorithm=] set to [=closeWritable=] with [=this=] as parameter, and [=WritableStream/set up/abortAlgorithm=] set to [=closeWritable=] with [=this=] as parameter.
+
+The <dfn>writeFrame</dfn> algorithm is given a |generator| and a |frame| as input. It is defined by running the following steps:
+1. If |generator|.{{MediaStreamTrack/kind}} equals `video` and |frame| is not a {{VideoFrame}} object, return [=a promise rejected with=] a {{TypeError}}.
+2. If |generator|.{{MediaStreamTrack/kind}} equals `audio` and |frame| is not an {{AudioData}} object, return [=a promise rejected with=] a {{TypeError}}.
+3. Send the media data backing |frame| to all live tracks connected to |generator|, possibly including |generator| itself.
```

Comment:
Do we need to mention anything about constraints here?

Reply:
No. Sources don't have constraints. Constraints belong to tracks.
We define track behavior (including constrainable properties) in the "Specialization of MediaStreamTrack behavior" section.

Reply:
My comment is based on this language in mediacapture-main: "When a source is connected to a track, it must produce media that conforms to the constraints present on that track, to that track." https://w3c.github.io/mediacapture-main/#dfn-source

Reply:
Also note the following language in mediacapture-main: "The media MUST NOT be upscaled, stretched or have fake data created that did not occur in the input source."